### PR TITLE
[extractor/boxcast] Add extractor

### LINKED ISSUE #5769: BoxCast.tv - not working.
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.11.11** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

USA

### Provide a description that is worded well enough to be understood

Embedded video.  Link of the embed is:
https://boxcast.tv/view-embed/in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx?showTitle=0&showDescription=0&showHighlights=0&showRelated=0&showCountdown=1&market=smb&showDocuments=0&showIndex=0&showDonations=0

Original website link:
https://childrenshealthdefense.eu/live-stream/

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://boxcast.tv/view-embed/in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx?showTitle=0&showDescription=0&showHighlights=0&showRe[2/2]
=0&showCountdown=1&market=smb&showDocuments=0&showIndex=0&showDonations=0']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.11.11 [8b64402] (pip)
[debug] Python 3.10.6 (CPython x86_64 64bit) - Linux-5.15.0-56-generic-x86_64-with-glibc2.35 (OpenSSL 3.0.2 15 Mar 2022, glibc 2.35)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.09.24, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1723 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Latest version: 2022.11.11, Current version: 2022.11.11
yt-dlp is up to date (2022.11.11)
[debug] [generic] Extracting URL: https://boxcast.tv/view-embed/in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx?showTitle=0&showDescription=0&showHighlights=0&showRelated=0&showCountdown=1&market=smb&showDocuments=0&showIndex=0&showDonations=0
[generic] in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx?showTitle=0&showDescription=0&showHighlights=0&showRelated=0&showCountdown=1&market=smb&showDocuments=0&showIndex=0&showDonations=0: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx?showTitle=0&showDescription=0&showHighlights=0&showRelated=0&showCountdown=1&market=smb&showDocuments=0&showIndex=0&showDonations=0: Extracting information
[debug] Looking for embeds
[debug] Identified a twitter:player iframe
[debug] [generic] Extracting URL: https://boxcast.tv/view-embed/in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx
[generic] in-the-midst-of-darkness-light-prevails-an-interdisciplinary-symposium-ozmq5eclj50ujl4bmpwx: Downloading webpage
ERROR: [generic] Unable to download webpage: Remote end closed connection without response (caused by RemoteDisconnected('Remote end closed connection without response')); please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 674, in extract
    ie_result = self._real_extract(url)
  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/extractor/generic.py", line 2582, in _real_extract
    full_response = self._request_webpage(url, video_id, headers={
  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 825, in _request_webpage
    raise ExtractorError(errmsg, cause=err)

  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/extractor/common.py", line 807, in _request_webpage
    return self._downloader.urlopen(self._create_request(url_or_request, data, headers, query))
  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/YoutubeDL.py", line 3692, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.10/urllib/request.py", line 519, in open
    response = self._open(req, data)
  File "/usr/lib/python3.10/urllib/request.py", line 536, in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
  File "/usr/lib/python3.10/urllib/request.py", line 496, in _call_chain
    result = func(*args)
  File "/home/dev/.local/pipx/venvs/yt-dlp/lib/python3.10/site-packages/yt_dlp/utils.py", line 1527, in https_open
    return self.do_open(
  File "/usr/lib/python3.10/urllib/request.py", line 1352, in do_open
    r = h.getresponse()
  File "/usr/lib/python3.10/http/client.py", line 1374, in getresponse
    response.begin()
  File "/usr/lib/python3.10/http/client.py", line 318, in begin
    version, status, reason = self._read_status()
  File "/usr/lib/python3.10/http/client.py", line 287, in _read_status
    raise RemoteDisconnected("Remote end closed connection without"
http.client.RemoteDisconnected: Remote end closed connection without response
```

### PR DESCRIPTION
**Pull request by HobbyistDev**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This PR add site support for single video extraction in boxcast.tv. This PR also include embed extraction and `/channel/<channel_id>?b=<broadcast_id>` and `/video-portal/<some string>/<some_string>/<broadcast_id>` url extraction

Fixes #5769

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Fix or improvement to an extractor (Make sure to add/update tests)
- [x] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/boxcast.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    traverse_obj,
+    unified_timestamp
+)
+
+
+class BoxCastVideoIE(InfoExtractor):
+    _VALID_URL = r'''(?x)(?:
+                    https?://boxcast\.tv/
+                    (?:view-embed/|channel/\w+\?(?:[^/#&]+&?)?b=|video-portal/(?:\w+/?){2})
+                    (?P<id>[\w-]+))
```
Comment (pukkandan) [inline yt_dlp/extractor/boxcast.py:13]:
```suggestion
    _VALID_URL = r'''(?x)(
        https?://boxcast\.tv/(?:
            view-embed/|
            channel/\w+\?(?:[^#]+&)?b=|
            video-portal/(?:\w+/){2}
        )(?P<id>[\w-]+)
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/boxcast.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            broadcast_json_data = self._download_json(
+                f'https://api.boxcast.com/broadcasts/{display_id}', display_id)
+
+        return {
+            'id': str(broadcast_json_data['id']),
+            'title': (broadcast_json_data.get('name')
+                      or self._html_search_meta(['og:title', 'twitter:title'], webpage)),
+            'description': (broadcast_json_data.get('description')
+                            or self._html_search_meta(['og:description', 'twitter:description'], webpage)
+                            or None),
```
[the remark this answers is not in the document]
Comment (HobbyistDev) [reply yt_dlp/extractor/boxcast.py:102]: Sometimes the `description` return `''` and it didn't assumed as `None`, is there a better way to handle this ?

[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/extractor/boxcast.py:95]: That's fine. Just making sure it was intentional

File: yt_dlp/extractor/boxcast.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            'uploader_id': 're8w0v8hohhvpqtbskpe',
+            'uploader': 'Children\'s Health Defense',
+        }
+    }]
+
+    def _real_extract(self, url):
+        display_id = self._match_id(url)
+        webpage = self._download_webpage(url, display_id)
+        webpage_json_data = self._search_json(
+            r'var\s*BOXCAST_PRELOAD\s*=', webpage, 'BOXCAST_PRELOAD', display_id,
```
Comment (pukkandan) [inline yt_dlp/extractor/boxcast.py:72]:
```suggestion
            r'var\s*BOXCAST_PRELOAD\s*=', webpage, 'broadcast data', display_id,
```

File: yt_dlp/extractor/boxcast.py [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+
+class BoxCastVideoIE(InfoExtractor):
+    _VALID_URL = r'''(?x)(
+            https?://boxcast\.tv/(?:
+                view-embed/|
+                channel/\w+\?(?:[^#]+&)?b=|
+                video-portal/(?:\w+/){2}
+            )(?P<id>[\w-]+))
+                '''
```
Comment (pukkandan) [inline yt_dlp/extractor/boxcast.py:16]:
```suggestion
    _VALID_URL = r'''(?x)
        https?://boxcast\.tv/(?:
            view-embed/|
            channel/\w+\?(?:[^#]+&)?b=|
            video-portal/(?:\w+/){2}
        )(?P<id>[\w-]+)'''
```

**Review by pukkandan — APPROVED** [review]

**Pull request merged**